Tests/E2E/MacTest.cs

```diff
@@ -204,5 +204,139 @@ private async Task Test_OTAA_Unconfirmed_Send_And_Receive_C2D_Mac_CommandsImplAs
             Assert.True(foundMacCommandReceivedMsg, $"Did not find in network server logs: '{macCommandReceivedMsg}'");
             Assert.True(foundDeviceMacCommandResponseMsg, $"Did not find in network server logs: '{deviceMacCommandResponseMsg}'");
         }
+
+        /// <summary>
+        /// Ensures that data rate is updated on a device after a LinkADRCmd MAC Command is sent as C2D message.
+        /// </summary>
+        [RetryFact]
+        public async Task Data_Rate_Is_Updated_When_C2D_With_LinkADRCmd_Received()
+        {
+            const int messagesToSend = 10;
+            const int warmUpMessageCount = 2;
+
+            var device = TestFixtureCi.Device32_ABP;
+            LogTestStart(device);
+
+            // Setup LoRa device properties
+            await ArduinoDevice.setDeviceModeAsync(LoRaArduinoSerial._device_mode_t.LWABP);
+            await ArduinoDevice.setIdAsync(device.DevAddr, device.DeviceID, device.AppEUI);
+            await ArduinoDevice.setKeyAsync(device.NwkSKey, device.AppSKey, device.AppKey);
+
+            // Setup protocol properties
+            // Start with DR5
+            await ArduinoDevice.SetupLora(TestFixtureCi.Configuration.LoraRegion, LoRaArduinoSerial._data_rate_t.DR5, 4, true);
+            await TestFixture.CleanupC2DDeviceQueueAsync(device.DeviceID);
+
+            // Sends 2x unconfirmed messages
+            for (var i = 1; i <= warmUpMessageCount; ++i)
+            {
+                var msg = PayloadGenerator.Next().ToString(CultureInfo.InvariantCulture);
+                Log($"{device.DeviceID}: Sending unconfirmed '{msg}' {i}/{messagesToSend}");
+                await ArduinoDevice.transferPacketAsync(msg, 10);
+
+                await Task.Delay(Constants.DELAY_BETWEEN_MESSAGES);
+                await AssertUtils.ContainsWithRetriesAsync("+MSG: Done", ArduinoDevice.SerialLogs);
+                TestFixture.ClearLogs();
+            }
+
+            var c2dMessage = new LoRaCloudToDeviceMessage()
+            {
+                Fport = 0,
+                Payload = string.Empty,
+                MacCommands = { new LinkADRRequest(3, 4, 25, 0, 1) } // Update data rate to DR3
```

Comment:
[nit] it would be clearer to understand what's going on here if we would name the arguments, e.g. `datarate: 3, ...`

Reply:
done in 4f479a77a0c2773553336b5d0a2b297d2bfa37d9